### DISCUSSION TITLE
Include Unicode symbols

### DISCUSSION THREAD

**Discussion by cbrnr** [quarto-dev/quarto-cli#937] [Q&A] [answered]
I'm trying to include the copyright symbol &#169; in my .qmd source file. When I insert the symbol in my text editor (VS Code), it does not get rendered at all. I can work around by writing `&copy;`, but I'd prefer to use the symbol directly.

On a related note, what if I wanted to use emojis? Those don't work either in the rendered output.

I'm using `format: html` and `engine: knitr` in case this is relevant.

**Comment by jjallaire** [comment]
Note sure what's going on here. I am able to include emjois and the copyright symbol and they both render:

![Screen Shot 2022-05-19 at 2 47 58 PM](https://user-images.githubusercontent.com/104391/169377364-d4814ae6-7d20-469c-9ab5-df8421213651.png)

**Reply by cbrnr** [reply]
Using the same input, I get this output:
<img width="1011" alt="Screen Shot 2022-05-20 at 07 37 00" src="https://user-images.githubusercontent.com/4377312/169458060-9cb41f6a-ecc6-4917-935b-07c8898987b3.png">

**Reply by cbrnr** [reply]
It is related to the engine. It doesn't work when I select `engine: knitr`, but it works correctly with `engine: jupyter`.

**Comment by mcanouil** [comment, marked as answer]
It is likely related to encoding issue, e.g., something that is not UTF-8.
Consider checking file encoding (status bar) and/or the global encoding setting in VSCode.

**Reply by cbrnr** [reply]
Unfortunately, this is not the case, I'm using UTF-8 (see screenshot above).

**Reply by mcanouil** [reply]
Could you share a repository with your example?
This would ensure, the file is exactly the same and not only in appearances.

**Reply by cbrnr** [reply]
https://github.com/cbrnr/quarto-unicode-emoji

It contains the exact file which produces the output in the screenshot above. Do you also want to see the generated HTML and other auxiliary files?

**Reply by mcanouil** [reply]
It renders properly.

I added R version and knitr version, maybe the issue is not on Quarto side? 🤔

<img width="1624" alt="image" src="https://user-images.githubusercontent.com/8896044/169472422-116f7c04-e36c-450d-ad0b-ed6725f5e5cb.png">

Does the emoji appear when using external browser to visualise the HTML file?

**Reply by cbrnr** [reply]
I'm using the latest versions:
![Screen Shot 2022-05-20 at 10 37 24](https://user-images.githubusercontent.com/4377312/169489149-fe0a3016-effa-4df5-93fc-9eb4e1c5ddd7.png)

The HTML renders the same in an external browser (no unicode, no emoji).

I have added the rendered files to the repository – maybe something can be gleaned by looking at the result?

**Reply by mcanouil** [reply]
This has to do with encoding for sure.
The copyright symbol encoding is `0xC2 0xA9` and in your HTML `<c2><a9></a9></c2>`.
To rule out VSCode, maybe try in Terminal (outside of VSCode): `quarto render unicode_emojis.qmd`.

**Reply by cbrnr** [reply]
No, it's exactly the same result. However, I tried rendering the exact same .qmd file from RStudio, and this works! Not sure if RStudio is using the same version or an integrated quarto, but I couldn't find the relevant information (I'm using a daily snapshot build of RStudio).

I tried several versions of quarto, including the latest (0.9.436) and the one you are using (0.9.414), but it doesn't work.

**Reply by jjallaire** [reply]
RStudio will use the version of Quarto on your PATH if there is one (so it should be using 0.9.426).

My guess is that VS Code is saving the file with a non UTF8 encoding (but RStudio always saves w/ UTF8 for qmd/rmd). There is likely some configuration in VS Code that is affecting this behavior.

**Reply by cbrnr** [reply]
> My guess is that VS Code is saving the file with a non UTF8 encoding (but RStudio always saves w/ UTF8 for qmd/rmd). There is likely some configuration in VS Code that is affecting this behavior.

I would be surprised if it didn't save the file as UTF-8. I've tried some other encodings, but they don't work either (as expected). And the source file shows the symbol and emoji correctly.

I open the exact same source file in VS Code and RStudio. Rendering in VS Code doesn't produce the symbols. Rendering in RStudio works, and switching back to VS Code Preview with the HTML rendered by RStudio quarto actually shows the symbols.

I've tried disabling all extensions but quarto, but no luck.

Any additional ideas I could try?

**Reply by cbrnr** [reply]
I just removed all user configuration files and extensions (`~/Library/Application Support/Code` and `~/.vscode`), which should give me the default VS Code settings. I installed the quarto extension, but it still didn't work.

**Reply by jjallaire** [reply]
quarto-cli reads and writes all text files as UTF-8. Pandoc also reads and writes all text files at UTF-8. The problem you are seeing is because some other piece of software is causing those symbols to be saved in a non-UTF8 encoding. What it is or why it is happening I can't say, but I can assure you its not quarto or the quarto vs code extension.

**Reply by mcanouil** [reply]
At this point,

- From Terminal -> no symbols
- From VSCode -> no symbols
- From RStudio -> symbols

might it be something related to your locale? 🤔

**Reply by cbrnr** [reply]
Thanks a lot @jjallaire and @mcanouil for your patience! My locale was really the reason why it didn't work! I set `LC_ALL="C"` to force English output in R, but that apparently also somehow influences the encoding. 🤯 So I guess I need to look for another way to stop getting localized R messages...
